e2e/registration_test.go

```diff
@@ -155,49 +155,68 @@ func TestE2E_ServiceRegistration_DeregisterWhenStopped(t *testing.T) {
 	assert.False(t, registered)
 }
 
-// TestE2E_ServiceRegistration_RegistrationFailure tests that if CTS is unable
-// to register itself as a service, it will not exit.
-func TestE2E_ServiceRegistration_RegistrationFailure(t *testing.T) {
+// TestE2E_ServiceRegistration_RegistrationSkipped tests scenarios where registration does
+// not happen, but CTS continues to run.
+func TestE2E_ServiceRegistration_RegistrationSkipped(t *testing.T) {
```

Comment:
The diff for this commit is hard to read, but I basically just put the old test in a for loop where the only difference between the test cases is the `service_registration` config. From a test perspective, all the setup and validation was the same whether the config disabled the registration or the config caused an error with registration.